src/main/java/net/neoforged/neoforge/common/util/AttributeUtil.java

```diff
@@ -0,0 +1,356 @@
+import net.neoforged.fml.loading.FMLEnvironment;
+import net.neoforged.neoforge.client.event.AddAttributeTooltipsEvent;
+import net.neoforged.neoforge.client.event.GatherSkippedAttributeTooltipsEvent;
+import net.neoforged.neoforge.common.NeoForge;
+import net.neoforged.neoforge.common.NeoForgeMod;
+import net.neoforged.neoforge.common.extensions.IAttributeExtension;
+import net.neoforged.neoforge.event.ItemAttributeModifierEvent;
+import net.neoforged.neoforge.internal.versions.neoforge.NeoForgeVersion;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+/**
+ * Utility code to support {@link IAttributeExtension}.
+ */
+public class AttributeUtil {
+    /**
+     * ID of the base modifier for Attack Damage
+     */
+    public static final ResourceLocation BASE_ATTACK_DAMAGE_ID = Item.BASE_ATTACK_DAMAGE_ID;
+
+    /**
+     * ID of the base modifier for Attack Speed
+     */
+    public static final ResourceLocation BASE_ATTACK_SPEED_ID = Item.BASE_ATTACK_SPEED_ID;
+
+    /**
+     * ID of the base modifier for Attack Range
+     */
+    public static final ResourceLocation BASE_ENTITY_REACH_ID = ResourceLocation.withDefaultNamespace("base_entity_reach");
+
+    /**
+     * ID used for attribute modifiers used to hold merged values when {@link NeoForgeMod#enableMergedAttributeTooltips()} is active.
+     * <p>
+     * Should not be used by any real attribute modifiers for gameplay purposes.
+     */
+    public static final ResourceLocation FAKE_MERGED_ID = ResourceLocation.fromNamespaceAndPath(NeoForgeVersion.MOD_ID, "fake_merged_modifier");
+
+    /**
+     * Comparator for {@link AttributeModifier}. First compares by operation, then amount, then the ID.
+     */
+    public static final Comparator<AttributeModifier> ATTRIBUTE_MODIFIER_COMPARATOR = Comparator.comparing(AttributeModifier::operation)
+            .thenComparing(Comparator.comparingDouble(a -> -Math.abs(a.amount()))) // Sort most impactful modifiers first
+            .thenComparing(Comparator.comparing(AttributeModifier::id));
+
+    private static final Logger LOGGER = LogManager.getLogger();
+
+    /**
+     * Checks if attribute modifier tooltips should show, and if they should, adds tooltips for all attribute modifiers present on an item stack to the stack's tooltip lines.
+     * <p>
+     * After the tooltip lines have been added, fires the {@link AddAttributeTooltipsEvent} to allow mods to add additional attribute-related lines.
+     * 
+     * @param tooltip A consumer to add the tooltip lines to.
+     * @param ctx     The tooltip context.
+     */
+    public static void addAttributeTooltips(ItemStack stack, Consumer<Component> tooltip, AttributeTooltipContext ctx) {
+        ItemAttributeModifiers modifiers = stack.getOrDefault(DataComponents.ATTRIBUTE_MODIFIERS, ItemAttributeModifiers.EMPTY);
+        if (modifiers.showInTooltip()) {
+            AttributeUtil.applyModifierTooltips(stack, tooltip, ctx);
```

Comment:
We are already in this class, we don't need to explicitly reference `AttributeUtil`

Reply:
I think these are holdovers from an automatic refactor when they got moved from `TooltipUtil`.  I'll clear those out.